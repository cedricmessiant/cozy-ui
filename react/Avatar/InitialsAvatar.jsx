import React from 'react'
import PropTypes from 'prop-types'

import Avatar, { avatarPropTypes } from '.'

const getInitials = name => {
  if (!name) return ''
  if (typeof name === 'string') {
    return name[0].toUpperCase()
  }

  return ['givenName', 'familyName']
    .map(part => name[part] || '')
    .map(name => name[0])
    .join('')
}

const InitialsAvatar = ({ name, ...rest }) => (
  <Avatar text={getInitials(name)} {...rest} />
)

InitialsAvatar.propTypes = {
  ...avatarPropTypes,
  name: PropTypes.oneOfType([
    PropTypes.shape({
      givenName: PropTypes.string,
      familyName: PropTypes.string
    }),
    PropTypes.string
  ]).isRequired
}

export default InitialsAvatar
