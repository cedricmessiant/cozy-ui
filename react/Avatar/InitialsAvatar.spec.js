import React from 'react'
import renderer from 'react-test-renderer'

import InitialsAvatar from './InitialsAvatar'

describe('InitialsAvatar', () => {
  it('should use the initial of the name (string)', () => {
    const name = 'kevon.willms@example.com'
    const component = renderer.create(
      <InitialsAvatar name={name} size="small" />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should use the initials of the name (object)', () => {
    const name = {
      givenName: 'Kevon',
      familyName: 'Willms'
    }
    const component = renderer.create(
      <InitialsAvatar name={name} size="small" />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
