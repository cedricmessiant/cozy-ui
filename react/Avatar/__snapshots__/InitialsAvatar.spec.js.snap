// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialsAvatar should use the initial of the name (string) 1`] = `
<div
  className="styles__c-avatar___PpDI- styles__c-avatar--small___1Y_Pv"
  style={
    Object {
      "backgroundColor": "var(--emerald)",
      "color": "white",
    }
  }
>
  
  <span
    className="styles__c-avatar-initials___310qC"
  >
    K
  </span>
</div>
`;

exports[`InitialsAvatar should use the initials of the name (object) 1`] = `
<div
  className="styles__c-avatar___PpDI- styles__c-avatar--small___1Y_Pv"
  style={
    Object {
      "backgroundColor": "var(--blazeOrange)",
      "color": "white",
    }
  }
>
  
  <span
    className="styles__c-avatar-initials___310qC"
  >
    KW
  </span>
</div>
`;
